libkbfs/folder_branch_ops.go

```diff
@@ -880,40 +942,18 @@ func (fbo *folderBranchOps) getMDForReadHelper(
 // the journal is on a branch, it returns an error.
 func (fbo *folderBranchOps) getMostRecentFullyMergedMD(ctx context.Context) (
 	ImmutableRootMetadata, error) {
-	lState := makeFBOLockState()
-
-	jServer, err := GetJournalServer(fbo.config)
-	if err != nil {
-		// Journaling is disabled entirely, so use the local head.
-		return fbo.getMDForReadHelper(ctx, lState, mdReadNoIdentify)
-	}
-
-	jStatus, err := jServer.JournalStatus(fbo.id())
+	mergedRev, err := fbo.getJournalPredecessorRevision(ctx)
 	if err != nil {
-		// Journaling is disabled for this TLF, so use the local head.
-		// TODO: JournalStatus could return other errors (likely
-		// file/disk corruption) that indicate a real problem, so it
-		// might be nice to type those errors so we can distinguish
-		// them.
-		return fbo.getMDForReadHelper(ctx, lState, mdReadNoIdentify)
-	}
-
-	if jStatus.BranchID != NullBranchID.String() {
-		return ImmutableRootMetadata{},
-			errors.New("Cannot find most recent merged revision while staged")
+		return ImmutableRootMetadata{}, err
 	}
 
-	if jStatus.RevisionStart == MetadataRevisionUninitialized {
-		// The journal is empty, so the local head must be the most recent.
+	if mergedRev == MetadataRevisionUninitialized {
+		// No unflushed journal entries, so use the local head.
+		lState := makeFBOLockState()
```

Comment:
Ah, I see, you moved the code from here. Maybe you need a way to distinguish the case of there being only the initial rev in the journal, so you can treat it as an error here but not above?